drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c

```diff
@@ -371,6 +371,116 @@ void wifi_nrf_event_proc_cookie_rsp(void *vif_ctx,
 #endif /* CONFIG_NRF700X_STA_MODE */
 #endif /* !CONFIG_NRF700X_RADIO_TEST */
 
+
+
+void configure_tx_pwr_settings(struct nrf_wifi_tx_pwr_ctrl_params *tx_pwr_ctrl_params,
+				struct nrf_wifi_tx_pwr_ceil_params *tx_pwr_ceil_params)
+{
+	tx_pwr_ctrl_params->ant_gain_2g = CONFIG_NRF700X_ANT_GAIN_2G;
+	tx_pwr_ctrl_params->ant_gain_5g_band1 = CONFIG_NRF700X_ANT_GAIN_5G_BAND1;
+	tx_pwr_ctrl_params->ant_gain_5g_band2 = CONFIG_NRF700X_ANT_GAIN_5G_BAND2;
+	tx_pwr_ctrl_params->ant_gain_5g_band3 = CONFIG_NRF700X_ANT_GAIN_5G_BAND3;
+	tx_pwr_ctrl_params->band_edge_2g_lo = CONFIG_NRF700X_BAND_2G_LOWER_EDGE_BACKOFF;
+	tx_pwr_ctrl_params->band_edge_2g_hi = CONFIG_NRF700X_BAND_2G_UPPER_EDGE_BACKOFF;
+	tx_pwr_ctrl_params->band_edge_5g_unii_1_lo = CONFIG_NRF700X_BAND_UNII_1_LOWER_EDGE_BACKOFF;
+	tx_pwr_ctrl_params->band_edge_5g_unii_1_hi = CONFIG_NRF700X_BAND_UNII_1_UPPER_EDGE_BACKOFF;
+	tx_pwr_ctrl_params->band_edge_5g_unii_2a_lo =
+					CONFIG_NRF700X_BAND_UNII_2A_LOWER_EDGE_BACKOFF;
+	tx_pwr_ctrl_params->band_edge_5g_unii_2a_hi =
+					CONFIG_NRF700X_BAND_UNII_2A_UPPER_EDGE_BACKOFF;
+	tx_pwr_ctrl_params->band_edge_5g_unii_2c_lo =
+					CONFIG_NRF700X_BAND_UNII_2C_LOWER_EDGE_BACKOFF;
+	tx_pwr_ctrl_params->band_edge_5g_unii_2c_hi =
+					CONFIG_NRF700X_BAND_UNII_2C_UPPER_EDGE_BACKOFF;
+	tx_pwr_ctrl_params->band_edge_5g_unii_3_lo = CONFIG_NRF700X_BAND_UNII_3_LOWER_EDGE_BACKOFF;
+	tx_pwr_ctrl_params->band_edge_5g_unii_3_hi = CONFIG_NRF700X_BAND_UNII_3_UPPER_EDGE_BACKOFF;
+	tx_pwr_ctrl_params->band_edge_5g_unii_4_lo = CONFIG_NRF700X_BAND_UNII_4_LOWER_EDGE_BACKOFF;
+	tx_pwr_ctrl_params->band_edge_5g_unii_4_hi = CONFIG_NRF700X_BAND_UNII_4_UPPER_EDGE_BACKOFF;
+
+#if defined(CONFIG_BOARD_NRF7002DK_NRF7001_NRF5340_CPUAPP) || \
+defined(CONFIG_BOARD_NRF7002DK_NRF5340_CPUAPP)
+	#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf70_tx_power_ceiling), max_pwr_2g_dsss)
+		tx_pwr_ceil_params->max_pwr_2g_dsss =
+			DT_PROP(DT_NODELABEL(nrf70_tx_power_ceiling), max_pwr_2g_dsss);
+	#else
+		tx_pwr_ceil_params->max_pwr_2g_dsss = 0x54;
```

Comment:
to simplify alignment and code readability, you can have all the defaults in a separation func e.g., `set_tx_pwr_defaults` and then keep only can keep only the overrides from DTS i.e., `#if` only.

Reply:
Defined the new function and moved initializations.